cli/src/stake.rs

```diff
@@ -2541,10 +2551,18 @@ pub(crate) fn fetch_epoch_rewards(
     rpc_client: &RpcClient,
     address: &Pubkey,
     mut num_epochs: usize,
+    starting_epoch: Option<u64>,
 ) -> Result<Vec<CliEpochReward>, Box<dyn std::error::Error>> {
     let mut all_epoch_rewards = vec![];
     let epoch_schedule = rpc_client.get_epoch_schedule()?;
-    let mut rewards_epoch = rpc_client.get_epoch_info()?.epoch;
+    let mut rewards_epoch = if let Some(epoch) = starting_epoch {
+        epoch
+    } else {
+        rpc_client
+            .get_epoch_info()?
+            .epoch
+            .saturating_sub(num_epochs as u64)
```

Comment:
I don't quite understand why do we subtract `num_epochs` now (before we didn't). Before we subtract `1` in the loop before calling `get_inflation_reward` so maybe should be also here subtract `1`?

Reply:
Because the loop was proceeding backwards by epoch before always from the newest epoch. I reversed the direction to now start at the older epoch and then move forward.

Reply:
`num_epochs` has to be at least 1. So it will subtract 1 here in the default case.

Reply:
https://github.com/anza-xyz/agave/pull/3085/files#diff-fbba192c63b85ce8ad0d8e150e332cab4cded14250ad105b5e2dcbadfe2bb5feL749

Reply:
Agree, everything looks fine